File: 1). DSA/1). Data Structures/01). Array/C++/_023)_Trapping_Rain_Water.cpp
#include<iostream>
using namespace std;

int getWater(int arr[], int n)
{
    int res = 0;
    for(int i=1; i<n-1; i++)
    {
        int lmax = arr[i];
        for(int j=0; j<i; j++)
            lmax = max(lmax, arr[j]);

        int rmax = arr[i];
        for(int j=i+1; j<n; j++)
            rmax = max(rmax, arr[j]);

        res += (min(lmax, rmax) - arr[i]);
    }
    return res;
}

int main()
{
    int arr[] = {3, 0, 1, 2, 5}, n=5;
    cout<<"Maximum fill Boxes = "<<getWater(arr, n)<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_023)_Trapping_Rain_Water.dart
import 'dart:math';

int getWater(arr) {
  int res = 0;

  for (int i = 1; i < arr.length - 1; i++) {
    int lMax = arr[i];
    for (int j = 0; j < i; j++) lMax = max(lMax, arr[j]);
    int rMax = arr[i];
    for (int j = i + 1; j < arr.length; j++) rMax = max(rMax, arr[j]);

    res += min(lMax, rMax) - arr[i] as int;
  }

  return res;
}

void main(List<String> args) {
  var arr = [3, 0, 1, 2, 5];
  print('Maximum fill boxes = ${getWater(arr)}');
}
